src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,53 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String DELIMITER = ",";
+    private static final String LS = System.lineSeparator();
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int supplyTotal = 0;
+        int buyTotal = 0;
+        int result = 0;
 
+        try (BufferedReader br = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                if (line.isBlank()) {
+                    continue;
+                }
+                String[] parts = line.split(DELIMITER);
+                if (parts.length != 2) {
+                    continue;
+                }
+                String operation = parts[0].trim();
+                int amount = Integer.parseInt(parts[1].trim());
```

Parsing the amount uses `Integer.parseInt(parts[1].trim())` without handling potential parsing failures. This violates checklist item #12: "Do not ignore exceptions\nLeaving an empty catch block or `e.printStackTrace` here is a bad practice. Better re-throw `RuntimeException` with the original exception and some message in the parameters:\n```
catch (Exception e) {
    throw new RuntimeException("Can't read data from the file " + fileName, e);
}
```
[CHECKLIST ITEM #12]"

Wrap the parse in a try/catch and on failure rethrow a `RuntimeException` with a clear message referencing `fromFileName` and include the caught exception as the cause. This ensures malformed numeric data triggers a deterministic failure with useful diagnostics.